src/pages/WorkspacePage.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Navigate, useParams } from 'react-router-dom';
+import { setActiveWorkspace } from '@openshift/dynamic-plugin-sdk-utils';
+import { useWorkspaceInfo } from '../utils/workspace-context-utils';
+
+const WorkspacePage: React.FunctionComponent = () => {
+  const name = useParams()?.workspaceName;
+  const { workspace } = useWorkspaceInfo();
+  if (!workspace) {
+    setActiveWorkspace(name);
+  }
+
+  return <Navigate to={`/stonesoup/workspaces/${workspace}/applications`} replace />;
```

Comment:
Why is the user getting redirected to applications page if he comes to stonesoup with a workspace context? I would assume he land on default overview page but with the context of workspace from the URL. From there he can either go to create application or application list page.

Reply:
Workspace redirection happens only when the user hits `/stonesoup/workspaces` url or if they click on Applications link in the sidebar.